### PR TITLE
Wallet UX Issues

### PR DESCRIPTION
**Pull request by ashleendaly**
- Wallet address now no longer has missing 0 when being copied and used in the faucet.
- Network now reflects the connected network in the browser plugin
- When a user copies a wallet address, a tick icon is displayed instead of the text "Copied". This keeps the UI cleaners and avoids overlapping text with the wallet address.

### CODE REVIEW

File: plugin/src/components/Wallet/index.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+import { MdCopyAll, MdCheck } from 'react-icons/md'
 import { type Network } from '../../utils/constants'
 import { useCurrentExplorer } from '../ExplorerSelector'
 import { getExplorerUrl, trimStr } from '../../utils/utils'
-import { useAccount, useProvider } from '@starknet-react/core'
+import { useAccount, useProvider, useNetwork } from '@starknet-react/core'
 import ConnectModal from '../starknet/connect'
 import DisconnectModal from '../starknet/disconnect'
-import { getChainName } from '../../utils/starknet'
+import { correctWalletAddress, getChainName } from '../../utils/starknet'
```
Comment (satyambnsal) [inline plugin/src/components/Wallet/index.tsx:11]: minor: perhaps we change the method name to `formatWalletAddress` ?

File: plugin/src/utils/starknet.ts [hunk 2]
```diff
@@ -56,3 +56,18 @@ function parse (value: any): string {
   if (typeof value === 'string') return value
   return BigNumber.from(value).toString()
 }
+
+export const correctWalletAddress = (address: string | undefined): string => {
+  if (address === undefined) return ''
```
Comment (varex83) [inline plugin/src/utils/starknet.ts:61]: should we return "undefined" here as well?

**Review by varex83 — CHANGES_REQUESTED** [review]
- Can you also make this button in the manual account change to the checkmark?
image
- On my side, in "Remote Devnet" section button still shows "Copied", can we change it to the checkmark as well? 
image
- Copy button in wallet connection works fine
- Also, please check the "Update" buttons, some of them doesn't rotate when triggered
- Plugin crashed after changing from "Wallet" connection to the "Manual Account" with following browser error (we can move out of this PR in the separate issue) 
image

File: plugin/src/components/Wallet/index.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import { MdCopyAll, MdCheck } from 'react-icons/md'
 import { type Network } from '../../utils/constants'
 import { useCurrentExplorer } from '../ExplorerSelector'
 import { getExplorerUrl, trimStr } from '../../utils/utils'
-import { useAccount, useProvider } from '@starknet-react/core'
+import { useAccount, useProvider, useNetwork } from '@starknet-react/core'
 import ConnectModal from '../starknet/connect'
 import DisconnectModal from '../starknet/disconnect'
-import { getChainName } from '../../utils/starknet'
+import { foprmatWalletAddress, getChainName } from '../../utils/starknet'
```
Comment (varex83) [inline plugin/src/components/Wallet/index.tsx:11]: I think you have a typo here :D

**Review by varex83 — CHANGES_REQUESTED** [review]
GTG after fixing the typo

**Pull request merged**